tests/integration/test_logic.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2021-2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pytest
+
+import cunumeric as num
+
+
+def test_inf():
+    arr = [np.inf, -np.inf, np.nan, 0]
+    assert np.array_equal(np.isposinf(arr), num.isposinf(arr))
+    assert np.array_equal(np.isneginf(arr), num.isneginf(arr))
+
+
+INPUTS = (
+    [1.0, 2.0, 3.0],
+    [1.0 + 0j, 2.0 + 0j, 3.0 + 0j],
+    [1.0 + 1j, 2.0 + 0j, 3.0 + 1j],
+)
+
+
+@pytest.mark.parametrize("arr", INPUTS)
+def test_predicates(arr):
+    in_np = np.array(arr)
+    in_num = num.array(arr)
+
+    assert np.array_equal(np.isreal(in_np), num.isreal(in_num))
+    assert np.array_equal(np.iscomplex(in_np), num.iscomplex(in_num))
+
+
+@pytest.mark.parametrize("arr", INPUTS)
+def test_array_predicates(arr):
+    in_np = np.array(arr)
+    in_num = num.array(arr)
+
+    assert np.array_equal(np.isrealobj(in_np), num.isrealobj(in_num))
+    assert np.array_equal(np.isrealobj(in_np), num.isrealobj(in_np))
+    assert np.array_equal(np.iscomplexobj(in_np), num.iscomplexobj(in_num))
+    assert np.array_equal(np.iscomplexobj(in_np), num.iscomplexobj(in_np))
+
+
+def test_isscalar():
```

Comment:
Test some more scalar-like values here:
```
ones((1,1))
ones((1,))
ones(())
ones((3,)).sum()
```

Reply:
I guess I could add a couple of cases where we return "deferred" scalars, and use them as a way to document the deviation from NumPy.

Reply:
4ed41c7d3495222adc67aa91293a7bfb661e22b3